### PR TITLE
1.0.0 wasm

### PR DESCRIPTION
**Pull request by AlexeyVavilin**

### CODE REVIEW

File: ton_client/client/src/client/wasm_client_env.rs [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+    ) -> ClientResult<WebSocket> {
+        // Connect to a server
+        let ws = if let Some(Some(protocols)) = headers.map(|mut headers| headers.remove("Sec-WebSocket-Protocol")) {
+            web_sys::WebSocket::new_with_str(url, &protocols)
+        } else {
+            web_sys::WebSocket::new(url)
+        }
+        .map_err(|_| Error::websocket_connect_error(url, "cannot create websocket"))?;
+
+        let (receive_sink, receive_stream) = futures::channel::mpsc::channel(100);
```
Comment (melsomino) [inline ton_client/client/src/client/wasm_client_env.rs:102]: suggest to rename `on_message_sink`, `on_message_stream`

File: ton_client/client/src/client/client.rs [hunk 2]
```diff
@@ -61,41 +56,25 @@ Note that default values are used if parameters are omitted in config"#,
...
-                message_processing_timeout: net_config.message_processing_timeout,
-                message_retries_count: net_config.message_retries_count,
-                out_of_sync_threshold: net_config.out_of_sync_threshold,
-                server_address: net_config.server_address.clone(),
-                wait_for_timeout: net_config.wait_for_timeout,
-            };
-            let sdk_client = ton_sdk::NodeClient::new(sdk_config);
-            (Some(client), Some(sdk_client))
+            let client = NodeClient::new(net_config.clone(), env.clone());
+            // let sdk_config = ton_sdk::NetworkConfig {
```
Comment (melsomino) [inline ton_client/client/src/client/client.rs:60]: Whay here is this commented code?

**Review by melsomino — CHANGES_REQUESTED** [review]
See comments

File: ton_client/client/src/client/client.rs (hunk 2 above)
Comment (AlexeyVavilin) [reply ton_client/client/src/client/client.rs]: Removed

File: ton_client/client/src/client/wasm_client_env.rs (hunk 1 above)
Comment (AlexeyVavilin) [reply ton_client/client/src/client/wasm_client_env.rs]: Done

File: ton_client/client/src/client/wasm_client_env.rs [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+        let send_result_msg = move |result: ClientResult<String>| {
```
[the remark this answers is not in the document]
Comment (AlexeyVavilin) [reply ton_client/client/src/client/wasm_client_env.rs]: Done

File: ton_client/client/src/client/std_client_env.rs [hunk 4]
```diff
@@ -133,15 +159,15 @@ impl ClientEnv for StdClientEnv {
     }
 
     /// Closes websocket
-    async fn websocket_close(&self, _handle: u32) {}
+    pub async fn websocket_close(&self, _handle: u32) {}
```
[the remark this answers is not in the document]
Comment (AlexeyVavilin) [reply ton_client/client/src/client/std_client_env.rs]: Removed function as not used

File: ton_client/client/src/net/node_client.rs [hunk 5]
```diff
@@ -121,15 +121,15 @@ pub(crate) struct Subscription{
 
 pub(crate) struct NodeClient {
     config: NetworkConfig,
-    client_env: Arc<dyn ClientEnv + Send + Sync>,
+    client_env: Arc<ClientEnv>,
     data: tokio::sync::RwLock<Option<InitedClientData>>,
```
[the remark this answers is not in the document]
Comment (AlexeyVavilin) [reply ton_client/client/src/net/node_client.rs]: Done

**Review by melsomino — APPROVED** [review]

**Pull request merged**